### PR TITLE
Isolate broken tests

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
@frankschae can you look at the terminate callback ones? I am not entirely sure how those were supposed to be working. They flip a problem from (0, Inf) to (Inf, 0) but then fail because they start at Inf. Is there supposed to be a mechanism that flips it to start at the terminating point? How is that terminating point saved?

**Comment by frankschae** [comment]
Yup, exactly. It's in the construction of the `ODEAdjointProblem`
https://github.com/SciML/SciMLSensitivity.jl/blob/master/src/backsolve_adjoint.jl#L148-L154

**Comment by ChrisRackauckas** [comment]
Awesome, and that explains the broken tests.

File: test/steady_state.jl [hunk 1]
```diff
@@ -485,7 +471,7 @@ end
                                     u0, p)
         @test du0≈Zdu0 atol=1e-4
         @test dp≈Zdp atol=1e-4
-        Zdu0, Zdp = Zygote.gradient((u0, p) -> loss(u0, p, sensealg = BacksolveAdjoint()),
+        Zdu0, Zdp = Zygote.gradient((u0, p) -> loss(u0, p, sensealg = BacksolveAdjoint(autojacvec = EnzymeVJP())),
```
Comment (ChrisRackauckas) [inline test/steady_state.jl:474]:
```suggestion
        Zdu0, Zdp = Zygote.gradient((u0, p) -> loss(u0, p, sensealg = BacksolveAdjoint()),
```

**Pull request merged**